### PR TITLE
Add YAML snippet how to preserve your battery pack

### PR DESCRIPTION
**Pull request by TheNexter**
readme updated to include an explanation of automation to preserve battery life

Closes: #49

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -170,6 +170,59 @@ uart:
     direction: BOTH
 ```
 
+## Preserving your batteries
```
Comment (syssi) [inline README.md:173]: Could you move this to the `Goodies` section and use a sub-headline? (`### Preserving your batteries`)

File: README.md [hunk 2]
```diff
@@ -170,6 +170,59 @@ uart:
...
+
+If you wish to limit the charge of your battery without an external system like a home assistant, take inspiration from this yml :
```
Comment (syssi) [inline README.md:175]:
```suggestion
If you wish to limit the charge of your battery without an external system like a home assistant, take inspiration from this YAML snippet:
```

File: README.md [hunk 3]
```diff
@@ -170,6 +170,59 @@ uart:
...
+
+sensor:
+  - platform: jbd_bms_ble
+    jbd_bms_ble_id: bms0
+    battery_strings:
+      name: "${name} battery cells"
+    charging_power:
+      name: "${name} charging power"
+    state_of_charge:
+      name: "${name} battery pourcentage"
```
Comment (syssi) [inline README.md:203]:
```suggestion
      name: "${name} state of charge"
```

File: README.md [hunk 4]
```diff
@@ -170,6 +170,59 @@ uart:
...
+      id: battery_percentage
```
Comment (syssi) [inline README.md:204]:
```suggestion
      id: soc
```

File: README.md [hunk 5]
```diff
@@ -170,6 +170,59 @@ uart:
...
+      on_value:
+        then:
+          # Manage charge
+          - if:
+              condition:
+                lambda: 'return id(battery_percentage).state < 80 && id(charging).state == false;'
```
Comment (syssi) [inline README.md:210]:
```suggestion
                lambda: 'return id(soc).state < 80 && id(charging).state == false;'
```

File: README.md [hunk 6]
```diff
@@ -170,6 +170,59 @@ uart:
...
+              then:
+                - switch.turn_on: charging
+          - if:
+              condition:
+                lambda: 'return id(battery_percentage).state > 80 && id(charging).state == true;'
```
Comment (syssi) [inline README.md:215]:
```suggestion
                lambda: 'return id(soc).state > 80 && id(charging).state == true;'
```

File: README.md [hunk 7]
```diff
@@ -170,6 +170,59 @@ uart:
...
+              then:
+                - switch.turn_off: charging
+          # Disable deep_sleep to stop charge at perfect timming (81% goes down to 80% after charge stop)
+          - if:
+              condition:
+                lambda: 'return id(battery_percentage).state >= 75 && id(battery_percentage).state <= 81 && id(charging).state == true;'
```
Comment (syssi) [inline README.md:221]:
```suggestion
                lambda: 'return id(soc).state >= 75 && id(soc).state <= 81 && id(charging).state == true;'
```

May be move the last condition to the front? It's more lightweight to evaluate a bool.

**Comment by TheNexter** [comment]
I think is good, if i miss something, sorry and say me what i need to change 😅

**Comment by syssi** [comment]
You could use ESPHome native methods instead of lambdas. Did you use the lambdas on purpose?

**Comment by TheNexter** [comment]
> You could use ESPHome native methods instead of lambdas. Did you use the lambdas on purpose?

This method allow the scooter / ebike disconnect from the ESP by going outside, comeback at what ever pourcentage you want and this method gonna work ;)

ESPhome do not understand the change between null (disconnected) and 62% (come back to home and connected) with the default automation method provide by ESPhome (on_value_range), you can trust me i try many time yesterday 😅

I have try at my office, this method work like a charm now ;)
Stop at 80% and enable charge if under 80%

**Comment by syssi** [comment]
I will provide another solution tomorrow. We can decide afterwards which solution should be added to the docs. It's hard to explain what I mean. :-)

**Comment by TheNexter** [comment]
> I will provide another solution tomorrow. We can decide afterwards which solution should be added to the docs. It's hard to explain what I mean. :-)

Oh ok ;)

**Comment by syssi** [comment]
What do you think about this approach: https://github.com/TheNexter/esphome-jbd-bms/blob/main/yaml-snippets/esp32-ble-deepsleep-limit-charging-automation.yaml

**Comment by syssi** [comment]
FYI I've added a `reboot_timeout: 0s` to some components (`wifi`, `api`, `mqtt`). Without this setting the ESPHome node will reboot every 15 minutes if there is no `wifi`, `api` or `mqtt` connection.

**Comment by TheNexter** [comment]
Oh your approach looks better than mine, your experience with esphome is clearly here 😄

I am gonna try your approach tomorrow ;)

**Pull request merged**